src/meson.build

```diff
@@ -367,6 +367,57 @@ executable(
   install_dir: daemon_dir
 )
 
+if libgbinder.found()
+  executable(
+    'fwupd-binder',
+    fwupdengine_rs,
+    fwupddaemon_rs,
+    plugins_hdr,
+    sources: [
+      daemon_loader_src,
+      'fu-daemon.c',
+      'fu-binder-daemon.c',
+    ],
+    include_directories: [
+      root_incdir,
+      fwupd_incdir,
+      fwupdplugin_incdir,
+    ],
+    dependencies: [
+      valgrind,
+      libsystemd,
+      engine_dep,
+      libgbinder,
+    ],
+    link_with: [
+      fwupdengine,
+      plugin_libs,
+    ],
+  )
+  fwupdmgr = executable(
+    'fwupdmgr-binder',
+    sources: [
+      'fu-binder-client.c',
+      'fu-debug.c',
+    ],
```

Comment:
For the client support:

When we first talked about this; I envisioned this being abstracted by `FwupdClient`.  The idea being ANY fwupd client can run on any OS and doesn't need to mind the communication mechanism behind it.  In my mind I should be able to take a client built with gobject introspection and run if the right introspection data is there run it on Android.

Is that abstraction a bad idea?

Reply:
So, I pondered this too -- e.g. a FwupdBinderClient as a superclass of FwupdClient. The more I thought about it the less I liked it. Android isn't going to use curl to download files -- and it's not even going to use a shared library to deploy the update. The "android way" is to just use binder, probably wrapped in a java interface -- rather than linking in shared libraries. Then I pondered "well, what about fwupdmgr" but we really don't want *all* the interface in binder -- and we probably want a few things split up -- for instance there's no property interface and signals act in a different way -- I came to the conclusion we probably only want to abstract out the client side bits when we know how the server bits are going to look. And abstracting FwupdClient with a couple of dozen vfuncs is a ton of work for a fwupdmgr that's never going to be used anyway.

Reply:
Got it; thanks for clarifying.  The unfortunate thing is this is going to mean the android client and everything else are going to divert quite a bit.

I guess on android for development purposes to keep binder and the lack of features in the client out of the equation `fwupdtool` can still be used though, right?

Reply:
> fwupdtool can still be used though, right

Yup, that's the plan -- it'll just use the engine as before.